dandiapi/api/services/metadata/__init__.py

```diff
@@ -129,36 +129,33 @@ def _build_validatable_version_metadata(version: Version) -> dict:
     with transaction.atomic():
         # validating version metadata needs to lock the version to avoid racing with
         # other modifications e.g. aggregate_assets_summary.
-        version = (
-            Version.objects.filter(id=version.id, status=Version.Status.PENDING)
-            .select_for_update()
-            .first()
-        )
-        version.status = Version.Status.VALIDATING
-        version.save()
+        version_qs = Version.objects.filter(id=version.id).select_for_update()
+        if not version_qs.filter(status=Version.Status.PENDING).exists():
+            # TODO: Should raise an error here?
```

Comment:
I'd be happy to just insert a log statement and return here, but I wasn't sure if this warranted raising a more concrete error.